clients/fides-js/src/lib/tcf/utils.ts

```diff
@@ -0,0 +1,37 @@
+import { PrivacyExperience } from "../consent-types";
+import { EnabledIds, TcfCookieConsent, TcfCookieKeyConsent } from "./types";
+import { TCF_KEY_MAP } from "./constants";
+import { generateTcString } from "../tcf";
+
+export const generateTcStringFromCookieTcfConsent = async (
+  experience: PrivacyExperience,
+  tcfConsent: TcfCookieConsent
+): Promise<string> => {
+  const enabledIds: any | EnabledIds = {
+    purposesConsent: [],
+    purposesLegint: [],
+    specialPurposes: [],
+    features: [],
+    specialFeatures: [],
+    vendorsConsent: [],
+    vendorsLegint: [],
+  };
+
+  TCF_KEY_MAP.forEach(({ cookieKey, enabledIdsKey }) => {
+    if (cookieKey) {
```

Comment:
`cookieKey` is always defined

Reply:
same Q as above- https://github.com/ethyca/fides/pull/4262#discussion_r1358129434 where we'd possibly have items within the `TCF_KEY_MAP` in the future that don't contain a `cookieKey`